include/albatross/src/utils/block_utils.hpp

```diff
@@ -52,6 +52,53 @@ struct BlockDiagonal {
   Eigen::MatrixXd toDense() const;
 };
 
+struct BlockSymmetric {
+
+  /*
+   * Stores a covariance matrix which takes the form:
+   *
+   *   X = |A   B|
+   *       |B.T C|
+   *
+   * It is assumes that both A and C - B.T A^-1 B are invertible and is
+   * designed for the situation where A is larger than C.  The primary
+   * use case is for a situation where you have a pre computed LDLT of
+   * a submatrix (A) and you'd like to perform a solve of the larger
+   * matrix (X)
+   */
+
+  BlockSymmetric(){};
+
+  BlockSymmetric(const Eigen::SerializableLDLT &A_, const Eigen::MatrixXd &B_,
```

Comment:
It's not clear to me based off the constructor arguments and comment what `S` is. It's only when looking at both the comment and the next constructor that this becomes clear. Should this constructor be private? It seems easy to use incorrectly.

Reply:
This is actually the constructor we'll end up using: (https://github.com/swift-nav/albatross/pull/144/files#diff-c7c355d66e5a4b694f01b0f3aaf0ed9eR497) so I don't want to make it private but could make the other one private (though that'll mess with tests) and/or add docs?